src/Simulation/Common/QubitManager.cs

```diff
@@ -17,22 +16,35 @@ namespace Microsoft.Quantum.Simulation.Common
     /// Allocation and release are O(1) operations.
     /// QubitManager uses memory - sizeof(long)*qubitCapacity. qubitCapacity is passed in to the constructor.
     /// </summary>
-    public class QubitManager
+    public class QubitManager : IQubitManager
     {
-        long NumQubits; // Total number of qubits this QubitManager is capable of handling.
-        long None; // Indicates the end of the list of free qubits. If we reach it - we are out of qubits.
-        long Allocated; // All qubits allocated by user will be marked with this number.
-        long Disabled; // All qubits disabled via DisableQubit interface will be marked with this number.
-        long AllocatedForBorrowing; // All qubits allocated only for borrowing, will be marked with this number or higher.
-        long[] qubits; // Tracks the allocation state of all qubits.
-        long free; // Points to the first free (unallocated) qubit.
-        long freeTail; // Points to the last free (unallocated) qubit. Only valid iff (!EncourageReuse).
-        long numAllocatedQubits; // Tracking this for optimization.
-        long numDisabledQubits; // Number of disabled qubits.
+        /// <summary>
+        /// Total number of qubits this QubitManager is capable of handling.
+        /// </summary>
+        protected long NumQubits;
+        /// <summary>
+        /// The number of currently allocated qubits.
+        /// </summary>
+        protected long NumAllocatedQubits;
+        /// <summary>
+        /// The number of disabled qubits.
+        /// </summary>
+        protected long NumDisabledQubits;
+        /// <summary>
+        /// Tracks the allocation state of all qubits.
+        /// </summary>
+        protected long[] qubits; 
```

Comment:
Based on the discussion below about growing this, would `List<long>` be better for growing capacity dynamically?

Reply:
The intent is that growing should happen fairly infrequently (it is doubling size every time, and that is only if MayExtendCapacity is explicitly enabled).